nucypher/blockchain/eth/domains.py

```diff
@@ -0,0 +1,55 @@
+from enum import Enum
+from typing import NamedTuple
+
+
+class ChainInfo(NamedTuple):
+    id: int
+    name: str
+
+
+class EthChain(ChainInfo, Enum):
```

Comment:
how come this is a class not a variable? There'll only be one instance right? or is it useful to have multiple `EthChain`s around?

Reply:
One instance _per ethereum chain_ . 

It's an enum - so it can only be specific values: MAINNET, GOERLI, SEPOLIA.

Reply:
It also adds some semantic value ie `EthChain.MAINNET` vs. `PolygonChain.MAINNET`

Reply:
sounds good! Thank you :)